util/db/cluster.go

```diff
@@ -60,16 +60,27 @@ func (db *db) ListClusters(ctx context.Context) (*appv1.ClusterList, error) {
 	clusterList := appv1.ClusterList{
 		Items: make([]appv1.Cluster, 0),
 	}
+	settings, err := db.settingsMgr.GetSettings()
+	if err != nil {
+		return nil, err
+	}
+	inClusterEnabled := settings.InClusterEnabled
 	hasInClusterCredentials := false
 	for _, clusterSecret := range clusterSecrets {
 		cluster, err := secretToCluster(clusterSecret)
 		if err != nil {
 			log.Errorf("could not unmarshal cluster secret %s", clusterSecret.Name)
 			continue
 		}
-		clusterList.Items = append(clusterList.Items, *cluster)
 		if cluster.Server == appv1.KubernetesInternalAPIServerAddr {
-			hasInClusterCredentials = true
+			if inClusterEnabled {
+				hasInClusterCredentials = true
+				clusterList.Items = append(clusterList.Items, *cluster)
+			} else {
+				log.Errorf("failed to add cluster %q to cluster list: in-cluster server address is disabled in Argo CD settings", cluster.Name)
+			}
+		} else {
+			clusterList.Items = append(clusterList.Items, *cluster)
 		}
 	}
 	if !hasInClusterCredentials {
```

Comment:
This if check needs to consult if inClusterEnabled is turned off.

Reply:
Good catch! Updated.